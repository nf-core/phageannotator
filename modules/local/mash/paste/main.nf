process MASH_PASTE {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mash:2.3--hd3113c8_4':
        'biocontainers/mash:2.3--hd3113c8_4' }"

    input:
    tuple val(meta) , path(sketch1)
    tuple val(meta2), path(sketch2)

    output:
    tuple val(meta), path("*.msh"), emit: msh
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}-${meta2.id}"
    """
    mash \\
        paste \\
        ${prefix}\\
        $sketch1 \\
        $sketch2 \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mash: \$( mash --version )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}-${meta2.id}"
    """
    touch ${prefix}.msh

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mash: \$( mash --version )
    END_VERSIONS
    """
}
