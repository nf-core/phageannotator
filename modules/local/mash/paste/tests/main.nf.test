nextflow_process {

    name "Test Process: MASH_PASTE"
    script "../main.nf"
    process "MASH_PASTE"


    test("test1.msh & test2.msh") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test1'],
                    file(params.pipelines_testdata_base_path + 'modules/local/mash/paste/bacteroides_fragilis_contigs1.msh', checkIfExists: true)
                ]
                input[1] = [
                    [id: 'test2'],
                    file(params.pipelines_testdata_base_path + 'modules/local/mash/paste/bacteroides_fragilis_contigs2.msh', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test1.msh & test2.msh - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [id: 'test1'],
                    file(params.pipelines_testdata_base_path + 'modules/local/mash/paste/bacteroides_fragilis_contigs1.msh', checkIfExists: true)
                ]
                input[1] = [
                    [id: 'test2'],
                    file(params.pipelines_testdata_base_path + 'modules/local/mash/paste/bacteroides_fragilis_contigs2.msh', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
