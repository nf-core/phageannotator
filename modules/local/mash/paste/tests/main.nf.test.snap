{
    "test1.msh & test2.msh - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1-test2.msh:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ae26f75a540c8b80b0e4172b54f0e89d"
                ],
                "msh": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1-test2.msh:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ae26f75a540c8b80b0e4172b54f0e89d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-21T12:59:43.149503984"
    },
    "test1.msh & test2.msh": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1-test2.msh:md5,37a3fadd36ab7a39f099465da3e17430"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ae26f75a540c8b80b0e4172b54f0e89d"
                ],
                "msh": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1-test2.msh:md5,37a3fadd36ab7a39f099465da3e17430"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ae26f75a540c8b80b0e4172b54f0e89d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-21T12:59:34.949677201"
    }
}
